### DISCUSSION TITLE
Prevent overriding native props/methods

### DISCUSSION THREAD

**Discussion by seahindeniz** [eslint/eslint#15110] [Ideas]
Hi, has there been any discussion or suggestion about preventing native method assignment, before? I have searched but couldn't encounter.

Something that prevents these assignments:
```js
Number.isNaN = () => {};
Object.entries = () => {};
console.log = () => {};
```

I have searched but couldn't encounter. Let me know if this is worth to consider, please.
Thanks!

**Comment by aladdin-add** [comment]
is something like [no-extend-native](https://eslint.org/docs/rules/no-extend-native) you are looking for?

**Reply by seahindeniz** [reply]
I think it's not. Here, [demo](https://eslint.org/demo#eyJ0ZXh0IjoiTnVtYmVyLmlzTmFOID0gKCkgPT4ge307XG5PYmplY3QuZW50cmllcyA9ICgpID0+IHt9O1xuY29uc29sZS5sb2cgPSAoKSA9PiB7fTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Im5vLWV4dGVuZC1uYXRpdmUiOjJ9LCJlbnYiOnsiYnJvd3NlciI6dHJ1ZSwibm9kZSI6dHJ1ZX19fQ==)

```js
Number.prototype.isNaN = () => {}; // Number prototype is read only, properties should not be added. (no-extend-native)

Number.isNaN = () => {}; // considered as good, it shouldn't be
```

**Reply by aladdin-add** [reply]
well, thanks! it seems a good enhancement to me, can you file an issue?
